### DISCUSSION TITLE
how to use this in a program?

### DISCUSSION THREAD

**Discussion by Zohrdan** [coqui-ai/TTS#864] [General Q&A]
this might be a super stupid question but where is the documentation on how to code with this? 

like something like this for the google TTS:

from gtts import gTTS
import os

s = "this is a test!"
file = "file.mp3"

tts = gTTS(s, 'en')
tts.save(file)

**Comment by erogol** [comment]
It is not TTS. It is gTTS 

not related at all

**Comment by Zohrdan** [comment]
That's why I wrote "like something like this for the google TTS"

I'm looking for a way to integrate coqui-tts into a program so I gave an example of what I'm looking for, using another TTS so you could extrapolate what I'm looking for. 

Well I found at least the command line for example
tts --text "this is a test!" --model_name tts_models/en/ljspeech/tacotron2-DDC_ph --use_cuda TRUE --out_path output.wav


but as I said I'm looking more for a way to implemt this in to a python program importing coqui TTS as module:


_from tts import TTS

s = "this is a test!"
file = "file.wav"

tts = TTS(s, 'tts_models/en/ljspeech/tacotron2-DDC_ph','TRUE',file)_

And yes I know that the above does not work... I just want to know if there is a way to import tts as a module?

**Reply by Zohrdan** [reply]
the same issue like this guy https://github.com/coqui-ai/TTS/discussions/775

**Reply by Zohrdan** [reply]
I think I found almost what I'm looking for  

from TTS.bin.synthesize import main as qtts

qtts() 

what do I need to fill in to get the same resulte as 

tts --text "this is a test!" --model_name tts_models/en/ljspeech/tacotron2-DDC_ph --use_cuda TRUE --out_path output.wav

**Reply by TheBlackJ3sus** [reply]
> I think I found almost what I'm looking for
> 
> from TTS.bin.synthesize import main as qtts
> 
> qtts()
> 
> what do I need to fill in to get the same resulte as
> 
> tts --text "this is a test!" --model_name tts_models/en/ljspeech/tacotron2-DDC_ph --use_cuda TRUE --out_path output.wav

```
import TTS
from pathlib import Path
from TTS.utils.manage import ModelManager
from TTS.utils.synthesizer import Synthesizer


def DoTheVoiceThing(WordsToSay,OutputDir):
    path = Path(TTS.__file__).parent  / ".models.json"
    manager = ModelManager(path)

    speakers_file_path = None
    vocoder_path = None
    vocoder_config_path = None
    encoder_path = None
    encoder_config_path = None
    use_cuda = True
    
    model_path, config_path, model_item = manager.download_model('tts_models/en/vctk/vits')
    vocoder_name = model_item["default_vocoder"]

    synthesizer = Synthesizer(
            model_path,
            config_path,
            speakers_file_path,
            vocoder_path,
            vocoder_config_path,
            encoder_path,
            encoder_config_path,
            use_cuda,
        )

    wav = synthesizer.tts(WordsToSay, 'p261')
    synthesizer.save_wav(wav, OutputDir)
    
DoTheVoiceThing('Bing Bong','BongBing.wav')
```

Probably not the best way but look at the synthesize.py, slim it down, make it a function, add your arguments, bing bong.

**Comment by thorstenMueller** [comment]
Maybe my project **cTTS** is helpful in that case.
https://pypi.org/project/cTTS/
